app/questionnaire/questionnaire_schema.py

```diff
@@ -105,6 +105,34 @@ def when_rules_section_dependencies_by_section(
     ) -> ImmutableDict[str, set[str]]:
         return ImmutableDict(self._when_rules_section_dependencies_by_section)
 
+    @cached_property
+    def when_rules_section_dependencies_for_progress(
```

Comment:
We should add docs strings to explain why this is being done.

Reply:
Added 👍